cluster.go

```diff
@@ -666,18 +683,35 @@ func (c *clusterClient) DoMulti(ctx context.Context, multi ...Completed) []Redis
 
 	results := resultsp.Get(len(multi), len(multi))
 
+	attempts := 1
+
 retry:
+	retryErrCh := make(chan error, len(multi))
+
 	wg.Add(len(retries.m))
 	mu.Lock()
 	for cc, re := range retries.m {
 		delete(retries.m, cc)
-		go c.doretry(ctx, cc, results, retries, re, &mu, &wg)
+		go c.doretry(ctx, cc, results, retries, re, &mu, &wg, retryErrCh)
 	}
 	mu.Unlock()
 	wg.Wait()
 
+	close(retryErrCh)
+
 	if len(retries.m) != 0 {
-		goto retry
+		for retryableErr := range retryErrCh {
+			shouldRetry := c.retryHandler.WaitUntilNextRetry(ctx, attempts, retryableErr)
+			if shouldRetry {
+				attempts++
+				goto retry
+			}
+		}
+
+		if retries.Redirects.Load() > 0 {
+			retries.Redirects.Store(0) // reset
+			goto retry
+		}
```

Comment:
I think `if retries.Redirects.Load() > 0` should have a higher priority than `for retryableErr := range retryErrCh`.

If there is a redirect response, we should not delay the retry.

Reply:
[f04b433](https://github.com/redis/rueidis/pull/643/commits/f04b433a99cf7b644720617ea5e4c5457ad9882b)